activerecord/lib/active_record/database_configurations/connection_url_resolver.rb

```diff
@@ -80,6 +79,12 @@ def raw_config
           end
         end
 
+        def resolved_adapter
+          adapter = uri.scheme && @uri.scheme.tr("-", "_")
+          adapter = ActiveRecord.protocol_adapters[adapter] || adapter
```

Comment:
I'm wondering if we should check if `adapter` maps to a registered adapter before looking it up as a protocol?

It's very unlikely, but assuming someone registered an adapter called `mysql`, it should have precedence for backward compact concerns.

Not too sure if it's worth it though.